src/keylessBackup/KeylessBackupIntro.tsx

```diff
@@ -1,69 +1,97 @@
+import { NativeStackScreenProps } from '@react-navigation/native-stack'
 import React from 'react'
 import { Trans, useTranslation } from 'react-i18next'
 import { SafeAreaView, ScrollView, StyleSheet, Text, View } from 'react-native'
 import { KeylessBackupEvents } from 'src/analytics/Events'
 import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
+import BackButton from 'src/components/BackButton'
 import Button, { BtnSizes, BtnTypes } from 'src/components/Button'
+import CancelButton from 'src/components/CancelButton'
 import Card from 'src/components/Card'
 import TextButton from 'src/components/TextButton'
+import CustomHeader from 'src/components/header/CustomHeader'
 import EnvelopeIcon from 'src/keylessBackup/EnvelopeIcon'
 import SmartphoneIcon from 'src/keylessBackup/SmartphoneIcon'
 import { KeylessBackupFlow } from 'src/keylessBackup/types'
-import { headerWithBackButton } from 'src/navigator/Headers'
 import { navigate } from 'src/navigator/NavigationService'
 import { Screens } from 'src/navigator/Screens'
-import { default as Colors, default as colors } from 'src/styles/colors'
+import { StackParamList } from 'src/navigator/types'
+import { Colors } from 'src/styles/colors'
 import { typeScale } from 'src/styles/fonts'
 import { Spacing } from 'src/styles/styles'
-
-function onPressContinue() {
-  ValoraAnalytics.track(KeylessBackupEvents.cab_setup_continue)
-  navigate(Screens.SignInWithEmail, { keylessBackupFlow: KeylessBackupFlow.Setup })
-}
-
 function onPressRecoveryPhrase() {
   ValoraAnalytics.track(KeylessBackupEvents.cab_setup_recovery_phrase)
   navigate(Screens.BackupIntroduction)
 }
 
-function SetUpKeylessBackup() {
+type Props = NativeStackScreenProps<StackParamList, Screens.KeylessBackupIntro>
+
+function KeylessBackupIntro({ route }: Props) {
+  const { keylessBackupFlow } = route.params
+  const isSetup = keylessBackupFlow === KeylessBackupFlow.Setup
   const { t } = useTranslation()
+
   return (
     <SafeAreaView style={styles.container}>
+      <CustomHeader left={isSetup ? <BackButton /> : <CancelButton />} style={styles.header} />
```

Comment:
I believe [other cab screens](https://github.com/valora-inc/wallet/blob/main/src/keylessBackup/KeylessBackupPhoneInput.tsx#L129) use navigationOptions that renders based on props. Is there a rule on when we should prefer CustomHeader over the navigationOptions?
Also, where does the CancelButton navigate to? We also have a KeylessBackupCancelButton that's shared among all subsequent screens which navigate appropriately based on the flow.

Reply:
The reason I used a CustomHeader here is because i needed to use either cancel or back depending if in setup or restore flow, otherwise I think header options are preferred.

Reply:
The other screens also need access to whether it's setup / restore, it's accessed here https://github.com/valora-inc/wallet/blob/62ceef4e0205afb365d41c6256e58572261dc0fd/src/keylessBackup/KeylessBackupPhoneInput.tsx#L133. Also not sure if its related, the alignment of the header was off on android in @jh2oman 's demo today. Maybe related to using CustomHeader?

Reply:
I was able to switch the navigation to use `SafeAreaView` from `react-native` and the above comment to resolve this issue. There is the bigger question of if we should be using `SafeAreaView` from `react-native-safe-area-context` instead, but for consistency on these screens I think that should be handled all at once rather then screen by screen.